gui/layers/_vectors_layer/model.py

```diff
@@ -0,0 +1,615 @@
+                    mean_region_y = np.mean(region[:, :, 1]*kernel_mat[:, :, 1])
+                    output_mat[i, j, 0] = mean_region_x
+                    output_mat[i, j, 1] = mean_region_y
+
+            self._vectors = self._check_vector_type(
+                output_mat[x_offset:-x_offset - 1:x, y_offset:-y_offset - 1:y])
+
+    @property
+    def width(self) -> Union[int, float]:
+        """
+        width of the line in pixels
+            widths greater than 1px only guaranteed to work with "agg" method
+        :return: int or float line width
+        """
+        return self._width
+
+    @width.setter
+    def width(self, width: Union[int, float]) -> None:
+        self._width = width
+        self._refresh()
+
+    @property
+    def length(self) -> Union[int, float]:
+        return self._length
+
+    @length.setter
+    def length(self, length: Union[int, float]):
+        """
+        Change the length of all lines
+        :param magnitude: length multiplicative factor
+        :return: None
+        """
+        self._length = length
+
+        if self._default_length in self._len_observers:
+            self._default_length(length)
+        else:
+            for callback in self._len_observers:
+                callback(self._length)
+
+        self._refresh()
+
+    def length_bind_to(self, callback):
+        '''
+        register an observer to be notified upon changes to length
+        Removes the default method for length if an external method is registered
+        :param callback: function to call upon length changes
+        :return:
+        '''
+        if self._default_length in self._len_observers:
+            self._len_observers.remove(self._default_length)
+            print('removing default length')
+        self._len_observers.append(callback)
+
+    def _default_length(self, newlen: int):
+        '''
+        Default method for calculating vector lengths
+        Implemented ONLY for image-like vector data
+        :param newlen: new length
+        :return:
+        '''
+
+        if self._data_type == 'projection':
+            return None
+        elif self._data_type == 'matrix':
+            self._length = newlen
+            self._vectors = self._check_vector_type(self._original_data)
+
+    @property
+    def color(self) -> str:
+        """Color, ColorArray: color of the body of the marker
+        """
+        return self._color
+
+    @color.setter
+    def color(self, color: str) -> None:
+        self._color = color
+        self._refresh()
+
+    @property
+    def connector(self):
+        """
+        line connector.  One of _connector.types = "strip", "segments"
+        or can receive an ndarray.
+        :return: connector parameter
+        """
+        return self._connector
+
+    @connector.setter
+    def connector(self, connector_type: Union[str, np.ndarray]):
+        self._connector = connector_type
+        self._refresh()
+
+    @property
+    def mode(self):
```

Comment:
It's unclear to me if we need any of the mode stuff. Note that the `image` layer doesn't have it, and I feel like if you aren't going to have multiple buttons on the left hand side then there is no point. I'd probably delete it (and just delete the `controls.py` inside view too. I can't remember now what happens if the `_qt_controls` property is `None`, but I think it just get's ignored and that space is left blank (if that doesn't happen we should fix it so it does).

Unless there is something else you'd rather put on the left hand side like a slider? Or you can imagine more interactive buttons etc

Reply:
would it make sense to have averaging and length adjustments on the left side?  Or is the left side mostly for buttons and/or sliders?

I'm totally fine with deleting controls completely.

Reply:
I think we should just delete the controls completely for now. Can always add them back in if we see a need for special interactivity, but I think how you have the averaging and length adjustment with the other properties makes more sense right now